browser-switch/src/main/java/com/braintreepayments/browserswitch/BrowserSwitchResult.java

```diff
@@ -9,38 +11,46 @@
 
 public class BrowserSwitchResult {
 
-    final private int status;
-    final private String errorMessage;
-    final JSONObject requestMetadata;
+    private final int status;
+    private final Uri deepLinkUri;
+    private final BrowserSwitchRequest request;
 
     @Retention(RetentionPolicy.SOURCE)
-    @IntDef({ STATUS_OK, STATUS_CANCELED, STATUS_ERROR })
-    public @interface BrowserSwitchStatus {}
+    @IntDef({STATUS_SUCCESS, STATUS_CANCELED})
+    public @interface BrowserSwitchStatus {
+    }
 
-    public static final int STATUS_OK = 1;
+    public static final int STATUS_SUCCESS = 1;
     public static final int STATUS_CANCELED = 2;
-    public static final int STATUS_ERROR = 3;
```

Comment:
I take it there's no longer a possibility for error? Or the error state is just being handled differently?

Reply:
Yup still possible, but since the error occurs when the `AndroidManifest.xml` is misconfigured or when there is no web browser (extreme edge case), we decided to throw an error synchronously when calling `BrowserSwitch#start()`. This eliminated the need for a callback parameter on start, and also allowed us to remove the error state from `BrowserSwitchResult`. Start would need to be called successfully at some point in order to receive a result.

Reply:
Nice!